numba/core/runtime/context.py

```diff
@@ -372,7 +376,27 @@ def _call_incref_decref(self, builder, typ, value, funcname):
             # function be called with a NULL pointer?
             fn.args[0].add_attribute("noalias")
             fn.args[0].add_attribute("nocapture")
-            builder.call(fn, [mi])
+
+            trace_str = ""
+
+            if config.DEBUG_NRT and config.DEBUG_NRT_STACK_LIMIT:
+                trace = io.StringIO()
+                traceback.print_stack(limit=config.DEBUG_NRT_STACK_LIMIT+2, file=trace)
+                # The last two stack frames are `_call_incref_decref`
+                # and `traceback.print_stack` which we ignore.
+                clean_trace = trace.getvalue().split('\n')[:-5]
+
+                if clean_trace:
+                    trace_str += 'Traceback:' + '\n'.join(clean_trace)+'\n\n'
+
+            mod = builder.module
+            # Make global constant for format string
+            cstring = cgutils.voidptr_t
+            fmt_bytes = cgutils.make_bytearray((trace_str + '\00').encode('ascii'))
+            global_fmt = cgutils.global_constant(mod, "printf_format", fmt_bytes)
```

Comment:
```suggestion
            global_fmt = cgutils.global_constant(mod, "nrt_debug_printf_format", fmt_bytes)
```
suggest this to make it more clear in the resulting LLVM IR what the global is for.

Reply:
Addressed in 170f84f